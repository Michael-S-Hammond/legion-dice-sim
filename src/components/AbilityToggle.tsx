import React from 'react';

type AbilityProps = {
    id: string,
    label: string,
    visible: boolean,
    active: boolean,
    onActiveChanged: (active: boolean) => void,
}

function AbilityToggle(props: AbilityProps) : JSX.Element {
    const handleActiveChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
        const newActive = e.target.checked;
        props.onActiveChanged(newActive);
    }

    return (
        <div key={`${props.id}-abilitytoggle`} className={`${ props.visible ? 'd-flex collapse.show' : 'collapse'} justify-content-center my-3 custom-control custom-switch`}>
            <input key={`${props.id}-abilitytoggle-input`} type="checkbox" className="custom-control-input my-auto"
                id={props.id}
                checked={props.active}
                onChange={handleActiveChanged}></input>
            <label key={`${props.id}-abilitytoggle-label`} className="custom-control-label drop-down-label mx-2 my-auto" htmlFor={props.id}>{props.label}</label>
        </div>
    );
}

export default AbilityToggle;
