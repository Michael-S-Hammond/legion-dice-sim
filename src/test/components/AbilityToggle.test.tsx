import React from 'react';

import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import AbilityToggle from '../../components/AbilityToggle';

describe('AbilityToggle', () => {
    it('matches the snapshot', () => {
        const onActiveChanged = jest.fn();

        const counter = <AbilityToggle
                id='testAbility'
                label='Test'
                visible={true}
                active={true}
                onActiveChanged={onActiveChanged}
            ></AbilityToggle>;
        const snapshot = create(counter);
        expect(snapshot.toJSON()).toMatchSnapshot();
    });

    it('matches the snapshot when not visible', () => {
        const onActiveChanged = jest.fn();

        const counter = <AbilityToggle
                id='testAbility'
                label='Test'
                visible={false}
                active={true}
                onActiveChanged={onActiveChanged}
            ></AbilityToggle>;
        const snapshot = create(counter);
        expect(snapshot.toJSON()).toMatchSnapshot();
    });

    it('handles active changed', () => {
        let isActive = false;
        const onActiveChanged = jest.fn((newisActive) => {
            isActive = newisActive;
        });

        const wrapper = shallow(<AbilityToggle
                id='testAbility'
                label='Test'
                visible={true}
                active={false}
                onActiveChanged={onActiveChanged}
            ></AbilityToggle>);

        wrapper.find('.custom-control-input').simulate('change', { target: { checked: true }});
        expect(onActiveChanged).toHaveBeenCalledTimes(1);
        expect(isActive).toEqual(true);
    })
});
