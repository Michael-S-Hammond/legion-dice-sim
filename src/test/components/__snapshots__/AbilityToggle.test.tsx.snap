// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbilityToggle matches the snapshot 1`] = `
<div
  className="d-flex collapse.show justify-content-center my-3 custom-control custom-switch"
>
  <input
    checked={true}
    className="custom-control-input my-auto"
    id="testAbility"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="custom-control-label drop-down-label mx-2 my-auto"
    htmlFor="testAbility"
  >
    Test
  </label>
</div>
`;

exports[`AbilityToggle matches the snapshot when not visible 1`] = `
<div
  className="collapse justify-content-center my-3 custom-control custom-switch"
>
  <input
    checked={true}
    className="custom-control-input my-auto"
    id="testAbility"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="custom-control-label drop-down-label mx-2 my-auto"
    htmlFor="testAbility"
  >
    Test
  </label>
</div>
`;
